sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -1663,15 +1665,28 @@ ProgramManager::link(const std::vector<device_image_plain> &DeviceImages,
   for (const device &Dev : Devs)
     PIDevices.push_back(getSyclObjImpl(Dev)->getHandleRef());
 
+  std::string LinkOptionsStr;
+  for (const device_image_plain &DeviceImage : DeviceImages) {
+    const std::shared_ptr<device_image_impl> &InputImpl =
+        getSyclObjImpl(DeviceImage);
+    const char *DeviceLinkOptions =
+        InputImpl->get_bin_image_ref()->getLinkOptions();
+    if (DeviceLinkOptions != nullptr) {
+      std::string TemporaryStr(DeviceLinkOptions);
+      std::optional<std::string> CompileOpts;
+      applyOptionsFromImage(*CompileOpts, TemporaryStr,
+                            *(InputImpl->get_bin_image_ref()));
+      LinkOptionsStr += TemporaryStr;
```

Comment:
If I am reading `applyOptionsFromImage` correctly, it will append `getLinkOptions` of the supplied `RTDeviceBinaryImage` to `LinkOpts` so `TemporaryStr` would contain the linker options for each image twice. Please correct me if I am wrong.

Reply:
You’re right, but it should not be a problem. Compilers can generally handle that. On the contrary, parsing options on rt side is hard: `-cc1 -fsycl-device-only`